### PR TITLE
preserve data

### PR DESCRIPTION
**Pull request by mumbereh**
### Implemented the following

- [ ] - In  this final milestone of my your portfolio website, I  have saved the form data in the local storage of the browser. In the same way when the user reloads the page the data they filled out in the form will be preserved.
- [ ] - When the user changes the content of any input field, the data is saved to the local storage.
- [ ] - When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
- [ ] - I created a single JavaScript object with all the data from the entire form to save it in local storage.

### CODE REVIEW

**Review by topeogunleye — COMMENTED** [review]
Hi @mumbereh, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ⚠️

   - You have submitted a pair programming project with no contributions from your partner.  Kindly report any issue you had with your partner to the student's success team and they'll permit you to work solo or look for a suitable partner.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

>Invalid Code Review Request **does not count** into the code reviews limit.

File: index.js [hunk 1]
```diff
@@ -189,4 +189,31 @@ function handleFormSubmit(event) {
...
+}
+
+const form = document.getElementById('contact-form');
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  if (form.checkValidity() && validateEmail()) {
+    saveData();
+    form.reset();
+  }
+});
```
Comment (nahnahdev) [inline index.js]: - [ ] Great job so far @mumbereh However, I noticed that data is not being preserved in your form as expected :+1: 
- [ ] Per the project requirements, after filling out the form, and not submitting it, the entered values should be saved in the browser but currently, they are only saved once the form is submitted and even with that the contact form information isn't persistent to page reloads as a result the form is still empty even though the `formdata` object was saved in the local storage so kindly fix this. :+1:  
- [ ] Please also ensure that when a user changes the content of any input field, the changes are also saved into the local storage :100: 

[Screencast from 06-21-2023 11:41:05 AM.webm](https://github.com/mumbereh/my-portfolio/assets/102699958/9a9b9204-21e3-428e-96d0-798ebb798c85)

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @mumbereh and @Wineshuga , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- Linters well setup :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- Nice Design :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @mumbereh and @Wineshuga , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All requested changes implemented :heavy_check_mark: 
- Data persisting in the form as expected :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**